client/src/pages/Post/PostWrite.tsx

```diff
@@ -1,12 +1,116 @@
 /* eslint-disable no-console */
 // /* eslint-disable react-hooks/exhaustive-deps */
 /* eslint-disable no-alert */
+import { ChangeEvent } from 'react';
+import styled from 'styled-components';
+import Button from '../../components/Button';
+import CustomEditor from '../../components/CustomEditor';
+import TextInput from '../../components/TextInput';
+import axios from 'axios';
+import { useState } from 'react';
+
 
 const PostWrite = () => {
+  const [title, setTitle] = useState('');
+  const [content, setContent] = useState('');
+  const [price, setPrice] = useState('');
+  const [category, setCategory] = useState('');
+  const [location, setLocation] = useState('');
+  const [userId, setUserId] = useState('');
+  const [userName, setUserName] = useState('');
+  const [userImg, setUserImg] = useState('');
+
+  const onChangeTitle = (e: ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+  };
+
+  const onChangeContent = (e: ChangeEvent<HTMLInputElement>) => {
+    setContent(e.target.value);
+  };
+
+  const onChangePrice = (e: ChangeEvent<HTMLInputElement>) => {
+    setPrice(e.target.value);
+  };
+
+  const onChangeCategory = (e: ChangeEvent<HTMLInputElement>) => {
+    setCategory(e.target.value);
+  };
+
+  const onChangeLocation = (e: ChangeEvent<HTMLInputElement>) => {
+    setLocation(e.target.value);
+  };
+
+  const onChangeUserId = (e: ChangeEvent<HTMLInputElement>) => {
+    setUserId(e.target.value);
+  };
+
+  const onChangeUserName = (e: ChangeEvent<HTMLInputElement>) => {
+    setUserName(e.target.value);
+  };
+
+  const onChangeUserImg = (e: ChangeEvent<HTMLInputElement>) => {
+    setUserImg(e.target.value);
+  };
+
+  const onSubmit = (e: ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    axios.post('http://localhost:4000/posts', {
+      title: title,
+      content: content,
+      price: price,
+      category: category,
+      location: location,
+      userId: userId,
+      userName: userName,
+      userImg: userImg,
+    })
+      .then((res) => {
+        console.log(res);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
```

Comment:
postWrite 컴포넌트의 state들 ( useState로 만든 "상태"들 )은 post로 서로 연관이 있는것 같습니다. 
이럴 경우에 각각의 데이터별로 state를 만들기보다 하나의 객체만들어 이용하는것을 추천드립니다.


예시로 코드를 제시해드립니다. 이 방법이 조금 복잡하다면 우선 넘어가셔도 됩니다 ^^
```suggestion

interface Post {
  title: string;
  content: string;
  price: string | number;
  category: string;
  location: string;
  userId: string;
  userName: string;
  userImg: string
  }
const [post,setPost] = useState<Post>({
  title: "";
  content: "";
  price: "";
  category: "";
  location: "";
  userId: "";
  userName: "";
  userImg: "";
})

  const onChangePost = (e: ChangeEvent<HTMLInputElement>) => {
const inputName =   e.target.name (또는 id )
setPost({
  ...post,
  [inputName]: e.target.value
})
  };

  const onSubmit = (e: ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();
    axios.post('http://localhost:4000/posts', post)
      .then((res) => {
        console.log(res);
      })
      .catch((err) => {
        console.log(err);
      });
  };

```

Reply:
넵! 좀더 연구해보고 바꿔보겠습니다!